flink-ai-flow/ai_flow_plugins/job_plugins/bash/bash_job_plugin.py

```diff
@@ -0,0 +1,190 @@
+                for raw_line in iter(sub_process.stdout.readline, b''):
+                    line = raw_line.decode(v.output_encoding).rstrip()
+                    self.log.info("%s", line)
+
+                sub_process.wait()
+
+                self.log.info('Command exited with return code %s', sub_process.returncode)
+
+                if sub_process.returncode != 0:
+                    raise Exception('Bash command failed. The command returned a non-zero exit code {}.'
+                                    .format(sub_process.returncode))
+                handle.lines[k] = line
+        return handle.lines
+
+    def get_job_status(self, job_handle: JobHandle) -> Status:
+        handler: BashJobHandle = job_handle
+        for sub_process in handler.sub_process.values():
+            if sub_process is not None:
+                if sub_process.poll() is None:
+                    return Status.RUNNING
+                else:
+                    if sub_process.returncode != 0:
+                        return Status.FAILED
+            else:
+                return Status.INIT
+        return Status.FINISHED
+
+    def submit_job(self, job: Job, job_runtime_env: JobRuntimeEnv) -> JobHandle:
+        handler = BashJobHandle(job=job, job_execution=job_runtime_env.job_execution_info)
+        bash_job: BashJob = job
+        processor_file = os.path.join(job_runtime_env.generated_dir, bash_job.processors_file)
+        with open(processor_file, 'rb') as f:
+            processors: Dict = serialization_utils.deserialize(f.read())
+        for k, v in processors.items():
+            processor: BashProcessor = v
+            env = os.environ.copy()
+            if 'env' in job.job_config.properties:
+                env.update(job.job_config.properties.get('env'))
+            sub_process = self.submit_one_process(processor=processor,
+                                                  env=env,
+                                                  working_dir=job_runtime_env.working_dir)
+            handler.sub_process[k] = sub_process
+        handler.processors_file = processor_file
+        return handler
+
+    def stop_job(self, job_handle: JobHandle, job_runtime_env: JobRuntimeEnv = None):
+        handler: BashJobHandle = job_handle
+        processor_file = handler.processors_file
+        with open(processor_file, 'rb') as f:
+            processors: Dict = serialization_utils.deserialize(f.read())
+        for k, v in processors.items():
+            self.log.info('{} Output:'.format(k))
+            sub_process = handler.sub_process.get(k)
+            self.log.info('Sending SIGTERM signal to bash process group')
+            if sub_process and hasattr(sub_process, 'pid') and sub_process.poll() is None:
+                while sub_process.poll() is None:
+                    try:
+                        os.killpg(os.getpgid(sub_process.pid), signal.SIGTERM)
+                    except Exception:
+                        time.sleep(1)
```

Comment:
If killing the process group exception, it's recommend to log this excetpion for the pid.